plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -289,51 +291,33 @@ fn get_name_and_version(line: &str) -> (&str, &str) {
     (name, version)
 }
 
-fn get_config() -> Option<TedgeConfig> {
-    let config_dir = TEdgeConfigLocation::default();
-
-    match fs::read_to_string(config_dir.tedge_config_file_path()) {
-        Ok(content) => match toml::from_str(&content) {
-            Ok(config) => Some(config),
-            Err(err) => {
-                warn!(
-                    "Failed to parse {}: {}",
-                    config_dir.tedge_config_file_path(),
-                    err
-                );
-                None
-            }
-        },
-        Err(_) => None,
+fn get_config(config_dir: PathBuf) -> Option<TEdgeConfig> {
+    let tedge_config_location = TEdgeConfigLocation::from_custom_root(config_dir);
+
+    match TEdgeConfigRepository::new(tedge_config_location).load_new() {
+        Ok(config) => Some(config),
+        Err(err) => {
+            warn!("Failed to load TEdgeConfig: {}", err);
+            None
+        }
     }
 }
 
 fn main() {
-    // On usage error, the process exits with a status code of 1
-
-    let mut apt = match AptCli::try_parse() {
-        Ok(aptcli) => aptcli,
-        Err(_) => {
-            AptCli::command()
-                .print_help()
-                .expect("Failed to print usage help");
-            // re-write the clap exit_status from 2 to 1, if parse fails
-            std::process::exit(1)
-        }
-    };
+    let mut apt = AptCli::parse();
```

Comment:
This change does not look okay. Software plugin has a strict rule to which status code to return when an error occured. https://thin-edge.github.io/thin-edge.io/references/plugin-api/#input-output-and-errors
> 1: usage. The command arguments cannot be interpreted, and the command has not been launched.
That's why the previos code was implemented like that as the comment says.

Indeed, if you run `tedge-apt-plugin list --aaa`,
* returning `1` previously
* returning `2` now

It should return `1` as it's parser error.